SayTheirNames/Source/Dependency/Deep Linking/DeepLinkDetails.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  DeepLinkDetails.swift
+//  SayTheirNames
+//
+//  Copyright (c) 2020 Say Their Names Team (https://github.com/Say-Their-Name)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import UIKit
+
+public struct DeepLinkDetails {
```

Comment:
To be honest I spent quite a lot of time trying to understand what calls what and where the actual link interpretation happens - does each link type contain a reference to this and tries to parse itself via this? Still a bit confused 😅 🙈 
That together with no documentation and tests, makes it hard to maintain I think.

Besides adding documentation and tests, I can suggest slightly modifying the code to make it more understandable in my view:
- keep `DeepLinkDetails` purely declarative, describing the link 
- `DeepLinkHandler` seems to be a central place that does the actual checking which link we actually got. I think `func deepLink(matching context: UIOpenURLContext) -> DeepLink?`  fits there more than here.

Reply:
Done.